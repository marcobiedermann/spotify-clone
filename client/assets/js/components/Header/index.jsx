import PropTypes from 'prop-types';
import React from 'react';
import './style.css';

const Header = ({ children }) => (
  <header className="header">
    {children}
  </header>
);

Header.propTypes = {
  children: PropTypes.node,
};

Header.defaultProps = {
  children: null,
};

export default Header;
