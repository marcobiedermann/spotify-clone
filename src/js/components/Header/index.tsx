import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './style.module.css';

export interface HeaderProps {
  className?: string;
}

const Header: FC<HeaderProps> = (props) => {
  const { className, ...otherProps } = props;

  return <header className={classNames(className, styles.header)} {...otherProps} />;
};

export default Header;
