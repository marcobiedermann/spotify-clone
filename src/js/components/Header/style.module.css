.header {
  background-color: #121212;
  display: flex;
  justify-content: space-between;
  padding: 1em 2em;
  position: sticky;
  top: 0;
}
