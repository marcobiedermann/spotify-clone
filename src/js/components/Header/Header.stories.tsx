import type { Meta, StoryObj } from '@storybook/react';
import Header from '.';

const meta: Meta<typeof Header> = {
  component: Header,
  args: {
    children: 'Header',
  },
};

type Story = StoryObj<typeof Header>;

const Default: Story = {};

export { Default };
export default meta;
