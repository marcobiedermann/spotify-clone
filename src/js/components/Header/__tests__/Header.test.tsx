import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { expect, it } from 'vitest';
import * as stories from '../Header.stories';

const { Default } = composeStories(stories);

it('renders correctly', () => {
  const { container } = render(<Default />);

  expect(container.firstChild).toMatchSnapshot();
});
