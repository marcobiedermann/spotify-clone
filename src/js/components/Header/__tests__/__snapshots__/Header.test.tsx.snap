// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<header
  class="header"
>
  Header
</header>
`;
